src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,974 @@
+                   file.delete();
+                   info("Deleted Java class file: " + file);
+                } else if (file.isDirectory()) {
+                    cleanTargetDir(file);
+                }
+            }
+        }
+        if (outputDirectory.listFiles().length == 0){
+            outputDirectory.delete();
+        }
+    }
+
+    /**
+     * Register the parent directory and all sub-directories with the WatchService
+     * 
+     * @param start parent directory
+     * @param dir path of parent directory
+     * @param watcher WatchService
+     * @throws IOException unable to walk through file tree 
+     */
+    protected void registerAll(final Path start, final Path dir, final WatchService watcher) throws IOException {
+        // register directory and sub-directories
+        Files.walkFileTree(start, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                debug("Watching directory: " + dir.toString());
+                dir.register(watcher,
+                        new WatchEvent.Kind[] { StandardWatchEventKinds.ENTRY_MODIFY,
+                                StandardWatchEventKinds.ENTRY_DELETE, StandardWatchEventKinds.ENTRY_CREATE },
+                        SensitivityWatchEventModifier.HIGH);
+
+                return FileVisitResult.CONTINUE;
+            }
+
+        });
+    }
+    
+    /**
+     * Get the file from the configDirectory if it exists
+     * 
+     * @param file 
+     * @return file or null if it does not exist
+     */
+    protected File getFileFromConfigDirectory(String file) {
+        File f = new File(configDirectory, file);
+        if (configDirectory != null && f.exists()) {
+            return f;
+        }
+        return null;
+    }
+
+    /**
+     * Given the fileChanged delete the corresponding Java class
+     * 
+     * @param fileChanged Java file changed
+     * @param classesDir the directory for compiled classes
+     * @param compileSourceRoot the source directory for the Java classes
+     * @throws IOException unable to resolve canonical path
+     */
+    protected void deleteJavaFile(File fileChanged, File classesDir, File compileSourceRoot) throws IOException {
+        if (fileChanged.getName().endsWith(".java")) {
+            String fileName = fileChanged.getName().substring(0, fileChanged.getName().indexOf(".java"));
+            File parentFile = fileChanged.getParentFile();
+            String relPath = parentFile.getCanonicalPath()
+                    .substring(parentFile.getCanonicalPath().indexOf(compileSourceRoot.getCanonicalPath())
+                            + compileSourceRoot.getCanonicalPath().length())
+                    + "/" + fileName + ".class";
+            File targetFile = new File(classesDir.getCanonicalPath() + relPath);
+
+            if (targetFile.exists()) {
+                targetFile.delete();
```

Comment:
Probably should check the return boolean to see if the file was deleted successfully and print info accordingly.

Reply:
fixed in https://github.com/WASdev/ci.common/pull/52